### ISSUE TITLE
Unable to export to html using includes at arbitrary depth

### ISSUE THREAD

**Issue by prog110** [asciidoctor/asciidoctor-gradle-plugin#445]
I am trying to generate html & pdf version from the following tree using asciidoctor 2.3.0 version of gradle plugin

Here is the source folder structure

![1](https://user-images.githubusercontent.com/452518/64772698-b1f12300-d56e-11e9-8fe5-2f692fa2879b.png)

And here is expected file structure

![2](https://user-images.githubusercontent.com/452518/64772705-b4537d00-d56e-11e9-92d8-ca1e506ec660.png)

I created a sample project [here](https://github.com/thekalinga/asciidoctor-2-test). Can you help me why

1. The sources are not included
2. Fragment is not included
3. Why highlighting is not working with rogue highlighter

Posting the issue here based on the conversation with @ysb33r on twitter

https://twitter.com/thekalinga/status/1171768438450057217

**Comment by ysb33r** [comment]
Your sources are not being found, because they are requested relative to baseDir and by default this `project.rootDir`. For instance if you add

```
asciidoctor {
   baseDirFollowsSourceDir()
}
```

Now the baseDir will change to be `src/docs/asciidoc`. However this is still on elevel above you `index.adoc`, so you also need to change

```
asciidoctorj {
   attributes 'sourcedir', 'course/source'
}
```

With `_fragment.adoc` the above no longer works because it is an adoc include from another adoc. Now the includes are relative including adoc i.e. relative to `nested/course/index.adoc. In there you will need to use `../{sourcedir}` instead.

**Comment by thekalinga** [comment]
Thanks for your reply

Based on that I fixed source includes. Fragment include still fails

Also, rogue plugin highlight does not work (I'm told I am missing some style sheets, but I am not sure where & how)

**Comment by thekalinga** [comment]
Including of fragment fails

![image](https://user-images.githubusercontent.com/452518/64864798-8137e900-d655-11e9-8912-89ae1486d496.png)

No line highlight

![image](https://user-images.githubusercontent.com/452518/64864841-99a80380-d655-11e9-8e78-7c1a620c1226.png)

**Comment by ysb33r** [comment]
>  rogue plugin highlight does not work 

According to what I'm seeing above Rogue is working.

**Comment by ysb33r** [comment]
> Based on that I fixed source includes. Fragment include still fails

I tyis becasue `_fragment.adoc` will be included relative to `baseDir`. You need to include `nested/course/_fragment.adoc`.

**Comment by thekalinga** [comment]
Any idea how to get it highlighted in the html? As the style is missing for that row?

**Comment by ysb33r** [comment]
> Any idea how to get it highlighted in the html? As the style is missing for that row?

Not sure what you are saying. I can see the source highlighter working. Are you referring to your theme/style for Rouge not being used?

**Comment by thekalinga** [comment]
Yes.. How to apply that styling for `hll` class within asciidoctor. I know its a basic question, but I could not find an answer elsewhere

**Comment by thekalinga** [comment]
Just a sec.. I am refering to this

https://asciidoctor.org/docs/user-manual/#rouge

Will let you know incase if I need any help after that

Thanks

**Comment by thekalinga** [comment]
Hi,

Fixed almost all issues except one problem. Now published a newer version with all fixes

This example now contains

1. How to embed code snippet
2. How to embed adoc fragment
3. How to refer to local images
4. Watermark incase of pdf

But there is one problem tho

When I generate html, the image references that I have in generated html is wrong

But the same code works for pdf without a problem

And the build generates no errors either

Can you please tell me why image `src` path in generated html is wrong

Thanks

**Comment by thekalinga** [comment]
Is there some issue with my setup/is this a bug?

Thanks

**Comment by mojavelinux** [comment]
> Any idea how to get it highlighted in the html? As the style is missing for that row?

The question is about how to add styles for the `.hll` selector. You need to use docinfo to add styles for that. See https://asciidoctor.org/docs/user-manual/#docinfo-file

**Comment by thekalinga** [comment]
Thanks for that. I have already incorporated that into the test project I have shared previously

Now the problem is this

> Hi,
> 
> Published a newer version with all fixes
> 
> This example now contains
> 
> 1. How to embed code snippet
> 2. How to embed adoc fragment & embed code within fragments
> 3. How to refer to local images
> 4. Watermark incase of pdf
> 
> But there is one problem tho
> 
> When I generate html, the image references that I have in generated html is wrong
> 
> But the same code works for pdf without a problem
> 
> And the build generates no errors either
> 
> Can you please tell me why image `src` path in generated html is wrong
> 
> Thanks

**Comment by thekalinga** [comment]
@ysb33r Can you please let me know if the issue is with library (or) any settings as images are properly bundled with pdf, but references in html is wrong

**Comment by ysb33r** [comment]
@thekalinga Were your problem ever solved?

**Comment by ysb33r** [comment]
Closing this issue.
